ls1intum/Artemis
# Fix NullPointerException in canAccessParticipation

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemis-staging.ase.in.tum.de and https://artemistest2.ase.in.tum.de/ with ~50 exercise details pages.

### Motivation and Context
Fix occasional `NullPointerException` when checking participation access with `canAccessParticipation`. This relates to Sentry issue 
`ARTEMIS-3YH`, see https://sentry.ase.in.tum.de/share/issue/b0e500ac54f74df9b402bf9887acdbb1/.

### Description
Rewrite logic for fetching the exercise in case it is missing, we directly fetch the exercise now from the DB. An added benefit is, the code is simpler and more readable.

### Steps for Testing
1. Activate the developer tools in your browser and look into the network tab
2. Open programming exercise details of various exercises and after each check the network tab. (This does not happen all the time, I would recommend trying exercises with different: age, course, exam, ...)
3. There should be no server error related to the latest results or submissions.
4. It might be a good idea to check sentry as well that the linked issue did not occur during testing in the test environment.

### Review Progress





- Code Review
- Manual Tests

### Test Coverage

* ProgrammingExerciseParticipationService.java: 95%
  ![image](https://user-images.githubusercontent.com/11130248/132717571-5fa9e446-3762-495c-bd30-6854c7a3f3d1.png)
